examples/multisig/lib.rs

```diff
@@ -575,50 +605,46 @@ mod multisig {
         /// Remove the transaction identified by `trans_id` from `self.transactions`.
         /// Also removes all confirmation state associated with it.
         fn take_transaction(&mut self, trans_id: TransactionId) -> Option<Transaction> {
-            let transaction = self.transactions.take(trans_id);
+            let transaction = self.transactions.get(&trans_id);
             if transaction.is_some() {
-                self.clean_transaction_confirmations(trans_id);
+                self.transactions.remove(&trans_id);
+                let pos = self
+                    .transaction_list
+                    .transactions
+                    .iter()
+                    .position(|t| t == &trans_id)
+                    .expect("The transaction exists hence it must also be in the list.");
+                self.transaction_list.transactions.swap_remove(pos);
+                for owner in self.owners.iter() {
+                    self.confirmations.remove(&(trans_id, *owner));
+                }
+                self.confirmation_count.remove(&trans_id);
             }
             transaction
         }
 
         /// Remove all confirmation state associated with `owner`.
         /// Also adjusts the `self.confirmation_count` variable.
         fn clean_owner_confirmations(&mut self, owner: &AccountId) {
-            use ::ink_storage::collections::stash::Entry as StashEntry;
-            for (trans_id, _) in
-                self.transactions
-                    .entries()
-                    .enumerate()
-                    .filter_map(|(n, entry)| {
-                        match entry {
-                            StashEntry::Vacant(_) => None,
-                            StashEntry::Occupied(value) => Some((n as u32, value)),
-                        }
-                    })
-            {
-                if self.confirmations.take(&(trans_id, *owner)).is_some() {
-                    *self.confirmation_count.entry(trans_id).or_insert(0) += 1;
+            for trans_id in &self.transaction_list.transactions {
+                let key = (*trans_id, *owner);
+                if self.confirmations.get(&key).is_some() {
+                    self.confirmations.remove(&key);
+                    let mut count = self.confirmation_count.get(&trans_id).unwrap_or(0);
+                    count += 1;
```

Comment:
Hmm, seems it is a bug, it should be -1=)

Reply:
You are absolutely right. Thanks for catching this.